test/compiler/inference.jl

```diff
@@ -2776,3 +2776,10 @@ for badf in [getfield_const_typename_bad1, getfield_const_typename_bad2]
     @test code[end] === Core.ReturnNode()
     @test_throws TypeError badf()
 end
+
+# issue #37532
+@test Core.Compiler.intrinsic_nothrow(Core.bitcast, Any[Type{Ptr{Int}}, Int])
+@test Core.Compiler.intrinsic_nothrow(Core.bitcast, Any[Type{Ptr{T}} where T, Ptr])
+@test_broken !Core.Compiler.intrinsic_nothrow(Core.bitcast, Any[Type{Ptr}, Ptr])
```

Comment:
Do you have a fix for that in the pipeline? Otherwise, just adding `isconcretetype(ty)` to the condition might be like a viable band-aid? It pessimizes cases like the one above, but it used to throw for those before, so still an improvement.